TradeItIosTicketSDK2/TradeItDeviceManager.swift

```diff
@@ -0,0 +1,69 @@
+import LocalAuthentication
+
+class TradeItDeviceManager {
+    
+    func authenticateUserWithTouchId(onSuccess onSuccess: () -> Void, onFailure: () -> Void) {
+        var error: NSError?
+        let myLocalizedReasonString = "Authentication is required"
+        
+        let context = LAContext()
+        
+        guard context.canEvaluatePolicy(LAPolicy.DeviceOwnerAuthentication, error: &error) else {
+            print("touch id not available on this device: \(error)")
+            onSuccess()
+            return
+        }
+        
+        context.evaluatePolicy(LAPolicy.DeviceOwnerAuthentication,
+                               localizedReason: myLocalizedReasonString,
+                               reply: { (success: Bool, evaluationError: NSError?) -> Void in
+                                
+                                if success {
+                                    print("authenticateUserWithTouchId succeeded")
+                                    onSuccess()
+                                }
+                                else {
+                                    guard let error = evaluationError else {
+                                        print("authenticateUserWithTouchId failed")
+                                        onFailure()
+                                        return
+                                    }
+                                    
+                                    switch error.code {
```

Comment:
Not sure how do we want to handle these errors. In the old SDK if the user cancel the authentication, we take him to the link broker screen, but it doesn't make sense to me, we should just deny access to him for launching portfolio or trading?